includes/Core/Util/Reset.php

```diff
@@ -178,4 +192,35 @@ private function get_rest_routes() {
 			),
 		);
 	}
+
+	/**
+	 * Handles the reset admin action.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $nonce WP nonce for action.
+	 */
+	private function handle_reset_action( $nonce ) {
+		if ( empty( $nonce ) || ! wp_verify_nonce( $nonce, static::ACTION ) ) {
+			wp_die( esc_html__( 'Invalid nonce.', 'google-site-kit' ), 400 );
```

Comment:
The `Invalid nonce.` error message is very confusing and doesn't tell users how they can solve this problem. Let's update it to something like `Your request to reset Site Kit is expired. Please, try again.` - where `try again` is the newly generated reset link.

Reply:
I agree the messaging could be better here, but this is the same message we use in many other places for the same thing. Let's address this in a separate issue so we can improve it everywhere rather than just here.

Reply:
Created https://github.com/google/site-kit-wp/issues/3098 as this relates to a follow-up issue we were planning to open as well 👍